src/scripts/main.js

```diff
@@ -3,92 +3,94 @@
 };
 
-function fillGameField(state) {
-  state.forEach((row, i) => {
-    row.forEach((cell, j) => {
-      const currentCell = fieldCells[i][j];
-
-      currentCell.className = !cell
-        ? 'field-cell'
-        : `field-cell field-cell--${cell}`;
-      currentCell.innerHTML = !cell ? '' : cell;
-    });
-  });
-}
-
-function showMessage() {
-  const gameStatus = game.getStatus();
-
-  for (const key in messages) {
-    if (Object.hasOwnProperty.call(messages, key)) {
-      const message = messages[key];
-
-      if (message) {
-        message.classList.toggle('hidden', key !== gameStatus);
+const actions = {
+  idle: () => {
+    game.setStatus('playing');
+    actions.playing();
+    messages[0].classList.add('hidden');
+    messages[1].classList.add('hidden');
+    messages[2].classList.add('hidden');
+    messages[3].classList.remove('hidden');
+  },
+  playing: () => {
+    for (let line = 0; line < 4; line++) {
+      for (let column = 0; column < 4; column++) {
+        cells[line * 4 + column].textContent =
+          game.getState()[line][column] || '';
       }
     }
-  }
-}
+    setStyles();
+    score.textContent = game.getScore();
+  },
+  win: () => {
+    actions.playing();
+    messages[0].classList.add('hidden');
+    messages[1].classList.remove('hidden');
+    messages[2].classList.add('hidden');
+    messages[3].classList.add('hidden');
+  },
+  lose: () => {
+    actions.playing();
+    messages[0].classList.remove('hidden');
+    messages[1].classList.add('hidden');
+    messages[2].classList.add('hidden');
+    messages[3].classList.add('hidden');
+  },
+};
 
-function updateScore(score) {
-  gameScore.innerHTML = score;
+function bindState(arg) {
+  actions[arg]();
 }
 
-button.addEventListener('click', () => {
-  const buttonText = button.textContent;
-
-  if (buttonText === 'Start') {
-    game.start();
-    button.textContent = 'Restart';
-    button.classList.replace('start', 'restart');
-  } else {
-    game.restart();
-    updateScore(0);
-    button.textContent = 'Start';
-    button.classList.replace('restart', 'start');
+const setStyles = () => {
+  for (const cell of cells) {
+    cell.className = bgColorByNumber[cell.textContent]
+      ? 'field-cell ' + bgColorByNumber[cell.textContent]
+      : 'field-cell';
   }
+};
 
-  const state = game.getState();
-
-  fillGameField(state);
-  showMessage();
-});
-
-document.addEventListener('keydown', (keyEvent) => {
-  keyEvent.preventDefault();
-
-  if (game.getStatus() !== 'playing') {
-    return;
-  }
-
-  const moveActions = {
-    ArrowUp: game.moveUp,
-    ArrowDown: game.moveDown,
-    ArrowLeft: game.moveLeft,
-    ArrowRight: game.moveRight,
-  };
-
-  const action = moveActions[keyEvent.key];
-
-  if (action) {
-    action.call(game);
+// Event Listeners
+document.addEventListener('keydown', (eventKey) => {
+  if (game.getStatus() === 'playing') {
+    if (eventKey.key === 'ArrowDown') {
+      game.moveDown();
+      bindState(game.getStatus());
```

Comment:
you use `game.getStatus()` a lot of times here - save it into a variable in the start of listener function